packages/api/tests/lib/test-granule-remove-from-cmr.js

```diff
@@ -187,7 +187,77 @@ test.serial('unpublishGranule() succeeds with Dynamo and PG granule', async (t)
   const {
     dynamoGranule,
     pgGranule,
-  } = await unpublishGranule(t.context.knex, originalPgGranule);
+  } = await unpublishGranule({ knex: t.context.knex, pgGranuleRecord: originalPgGranule });
+
+  t.true(cmrDeleteStub.called);
+  t.deepEqual(
+    dynamoGranule,
+    omit(
+      {
+        ...originalDynamoGranule,
+        published: false,
+        updatedAt: dynamoGranule.updatedAt,
+      },
+      'cmrLink'
+    )
+  );
+  t.deepEqual(
+    pgGranule,
+    {
+      ...pgGranule,
+      published: false,
+      cmr_link: null,
+    }
+  );
+});
+
+test.serial('unpublishGranule() accepts an optional collection', async (t) => {
+  const { fakeCollection } = t.context;
+
+  const {
+    originalDynamoGranule,
+    originalPgGranule,
+    pgGranuleCumulusId,
+  } = await createGranuleInDynamoAndPG(t, {
+    published: true,
+    collectionId: constructCollectionId(fakeCollection.name, fakeCollection.version),
+  });
+
+  t.like(
+    await t.context.granulesModel.get({ granuleId: originalDynamoGranule.granuleId }),
+    {
+      published: true,
+      cmrLink: originalDynamoGranule.cmrLink,
+    }
+  );
+
+  t.like(
+    await t.context.granulePgModel.get(t.context.knex, {
+      cumulus_id: pgGranuleCumulusId,
+    }),
+    {
+      published: true,
+      cmr_link: originalPgGranule.cmr_link,
+    }
+  );
+
+  const cmrMetadataStub = sinon.stub(CMR.prototype, 'getGranuleMetadata').resolves({
+    foo: 'bar',
+  });
+  const cmrDeleteStub = sinon.stub(CMR.prototype, 'deleteGranule').resolves();
```

Comment:
See above - this stub is masking a failure in how the passed in collection is failing a lookup.

Reply:
Super unnerving. See above for explanation.